SMT/Overlay.xaml.cs

```diff
@@ -339,6 +339,17 @@ public Overlay(MainWindow mw)
             _ = DataOverlayUpdateLoop();
         }
 
+        /// <summary>
+        /// Calculates the actual size for the canvas element that represents a system. 
+        /// Takes into account if it is the current system the player is in.
+        /// </summary>
+        /// <param name="systemName">The name of the system for which the size should be calculated.</param>
+        /// <returns>The size for the canvas element for the system.</returns>
+        private float CalculatedOverlaySystemSize(string systemName)
+        {
+            return gathererMode ? overlaySystemSizeGatherer : (currentPlayerSystemData != null && currentPlayerSystemData.system != null && systemName == currentPlayerSystemData.system.Name ? overlaySystemSizeHunter * overlayCurrentSystemSizeHunterModifier : overlaySystemSizeHunter);
```

Comment:
Double inline ternary is a code smell and hard for some people (myself included) to read.

You should also be able to use Optional Chaining here instead of nesting multiple `a != null && a.b != null`. Something like this:

```cs
if (gathererMode) {
    return overlaySystemSizeGatherer;
}

if (systemName == currentPlayerSystemData?.system?.Name) {
    return overlaySystemSizeHunter * overlayCurrentSystemSizeHunterModifier;
}

return overlaySystemSizeHunter;
```

Reply:
monadic null checking has been in the language a while

Reply:
Yeah, the double ternary might be a bit too much but i prefer ternary to writing (multiple) if constructs. But you are right, for readability, this is a bit convoluted, noted for next time!
If monadic null checks are preferred i will use them in the future, i am not a fan of the readability, i tend to overlook that question mark too often.